frontend/src/Frontend/UI/Transfer.hs

```diff
@@ -678,61 +678,83 @@ transferDialog model netInfo ti ty fks tks unused = Workflow $ do
-    let backW = transferDialog model netInfo ti ty fks tks unused
+      (currentTab, _done) <- transferTabs newTab $ shouldWarningTabBeShown <$> tiDyn <*> meta
+      (conf, meta, payload, dSignedCmd, destChainInfo, tiDyn) <- mainSection currentTab
+      (cancel, newTab, next) <- footerSection currentTab meta dSignedCmd tiDyn
+    let backW ti = transferDialog model netInfo ti ty fks tks unused
     case destChainInfo of
       Nothing -> do
-        let nextScreen = ffor (tag (current dSignedCmd) next) $ \case
-              Nothing -> Workflow $ pure (mempty, never)
-              Just sc -> previewDialog model netInfo ti payload sc backW $
-                           sameChainTransferAndStatus model netInfo ti sc
+        let allDyns = (,)
+              <$> current tiDyn
+              <*> current dSignedCmd
+        let nextScreen = ffor (tag allDyns next) $ \case
+              (_, Nothing) -> Workflow $ pure (mempty, never)
+              (ti, Just sc) ->
+                previewDialog model netInfo ti payload sc (backW ti) $
+                  sameChainTransferAndStatus model netInfo ti sc
         pure ((conf, close <> cancel), nextScreen)
 
       Just (dgp, dss) -> do
-        let allDyns = (,,,,)
+        let allDyns = (,,,,,)
               <$> current payload
               <*> current dSignedCmd
               <*> current dgp
               <*> current dss
               <*> current meta
+              <*> current tiDyn
         let nextScreen = ffor (tag allDyns next) $ \case
-              (_,Nothing,_,_,_) -> Workflow $ pure (mempty, never)
-              (p,Just sc,gp,ss,meta') -> previewDialog model netInfo ti payload sc backW $
-                                           crossChainTransferAndStatus model netInfo ti sc gp ss meta'
+              (_,Nothing,_,_,_,_) -> Workflow $ pure (mempty, never)
+              (p,Just sc,gp,ss,meta',ti) ->
+                previewDialog model netInfo ti payload sc (backW ti) $
+                  crossChainTransferAndStatus model netInfo ti sc gp ss meta'
         pure ((conf, close <> cancel), nextScreen)
   where
+    shouldWarningTabBeShown currentTi currentMeta =
+      _ti_maxAmount currentTi && Just (_ca_account $ _ti_fromAccount currentTi) == _transferMeta_senderAccount currentMeta
     mainSection currentTab = elClass "div" "modal__main" $ do
-      (conf, meta, destChainInfo) <- tabPane mempty currentTab TransferTab_Metadata $
+      (conf, meta, destChainInfo, tiDyn) <- tabPane mempty currentTab TransferTab_Metadata $
         transferMetadata model netInfo fks tks ti ty
-      let payload = buildUnsignedCmd netInfo ti ty <$> meta
+      void $ tabPane mempty currentTab TransferTab_MaxTransactionWarning $ do
+        el "h2" $ text "You have asked for a max transfer but used the same account to pay gas. As such, the amount being transferred has changed."
+        el "div" $ do
+          el "span" $ text "Amount being transferred (after deducting gas): "
```

Comment:
Can we add an: "Original Amount" and "Gas Cost" in addition to "New Amount" ?

Reply:
Done.